File: src/SlidingWindow/Exercise1493.java
package SlidingWindow;
public class Exercise1493 {
    class Solution {
        public int longestSubarray(int[] nums) {
            
            int left = 0, right = 0;
            int count = 1;
            int max=0;
            while (right<nums.length) {
                if(nums[right]==0)
                    count--;
                
                if (count < 0) {
                    max = Math.max(max, right - left-1);
                    if (nums[left] == 0)
                        count++;
                    left++;
                }
                right++;
            }
            max = Math.max(max, right - left-1);
            return max;
        }
    }
}

File: src/SlidingWindow/Exercise1493.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    int longestSubarray(vector<int>& nums) {

        int left = 0, right = 0;
        int maxNum = 0;
        int count = 0;
        for (; right < nums.size();right++){
            if(nums[right]==0){
                count++;
            }
            if(count>1){
                if(nums[left]==0) {
                    count--;
                }
                left++;
            }
            maxNum = max(maxNum, right - left);
        }
        return maxNum;
    }
};

int main(){
    vector<int> nums{0, 1, 1, 1, 0, 1, 1, 0, 1};
    Solution sl;
    cout << sl.longestSubarray(nums);
}